### ISSUE TITLE
Problem with speed

### ISSUE THREAD

**Issue by wpsoul** [vafour/vafpress-framework#76]
I have a big problem with speed of metaboxes. I use it on premium theme. I have near 10 blocks and each block upload only on clicking image checkbox. But it's very slow, after click on checkbox it need near 4-5 seconds sometimes before block begins to load.

Also sometimes standart button of wordpress (Publish) become inactive http://joxi.ru/FvGGU_3JTJAZVDVBEWk. 

Also sorter field have php bugs, when I add it, I have error with undefinite index. On some hostings it causes white screen of death.

Awesome framework but need some update.

**Comment by Zackio** [comment]
I wonder if this is related to my problem, sometimes it's not just slow but never completes. Have you found a solution?

**Comment by wpsoul** [comment]
No, I can't find solution, and author didn't answer.

**Comment by Zackio** [comment]
I've been bug fixing and I found my problem is was due to the the javascript fade out and in of the loader and dependent field. The field fadein is being called before it had finished fading out. Server is too fast for the script. I'm going to use a deferred object to fix it.

**Comment by wpsoul** [comment]
If you will resolve problem, please, write here. I also found that problem is in dependent fields. If these fields is too much on page - it's very slowly

**Comment by Zackio** [comment]
The deferred method has worked. When I have finished and have cleaned up code I will post it. It's a great framework and I'm very greatful that it has been offered free but there's alot of bugs in the binding and dependencies that need ironing out. Hopefully I will be able to do that.

**Comment by wpsoul** [comment]
Yes, this is great framework. I tried all others, but this is most powerfull. It still needs many improvements, so maybe someone will continue develop it.

**Comment by Zackio** [comment]
I haven't tried many but I spent I long time comparing features and this seems the most powerful. I'm surprised more people aren't using it. The coding quality is very high.

I've forked with my fix, working well for me thus far. The speed issue, it's when lot's of items are on the page but fine when only a couple?

**Comment by wpsoul** [comment]
No, when only couple items, it's slowly but not critical, but when many items, it become very slowly (3-5 seconds for response)

**Comment by wpsoul** [comment]
Are there any news?

**Comment by Zackio** [comment]
Hi wpsoul, I haven't come across any slow down issues. it's all been working perfectly for me except I had the ajax never loading with dependency items like I did with binded which requited the same fix.

**Comment by wpsoul** [comment]
It's because you have only some fields. When you will enable something like page builder with more than 10 sortable group, you will find these slow down issues. I found that problem is in jquery scripts and dependency. For example, I have 10 groups and image checkbox with dependency. When I click to image one of the group must load and other must hide. But, the main problem - I click on image checkbox, loader image displays, then it hides, but after that I have 3-5 seconds pause before block appears and nothing happens in this 5 seconds.

**Comment by thednp** [comment]
I had a word with devs some time ago, the AJAX dependency functions cannot work with several fields at the same time or just too many. Yes the code as is, it's brilliant, it allows you to create your own dependencies and functions, but cannot be extended to the point of making a page builder as we know it today.

You would have to make an override for the controls and create new fields for the page builder you create with your own interactions.

**Comment by wpsoul** [comment]
But I have dependency functions that works only with one field. Logic is simple: there are image checkbox and several group of field. When click on image checkbox - one group is loading. Absolutely the same page builder is described in docs and in demo. And I found strange thing. When I install framework and create new page, everything works good, but after saving post, each time I edit post one more time, speed become slowly and slowly after each saving. I tried even purge revisions, but it doesn't help.

**Comment by thednp** [comment]
This is because that group, perhaps the builder module has a ton of fields inside. Having let's say 20 modules (repeatable groups) and creating a bunch of them for an example page, the current code is gonna go very slow or even block.

I developed a very powerful slider with tons of options and have encountered this speed issue as well. I decided (as suggested by VF devs) to split these options for 2 post types, and now I have speed, guaranteed not blocking anywhere.

For a page builder, I would suggest to find a better interface, I would personally go for Live Composer, develop your own modules and the outcome is just way better.

**Comment by wpsoul** [comment]
Yes, I think about it. I tried visual composer plugin, but it's good for shortcodes, but bad for content blocks. Now, I look to LC and IG pagebuilder

**Comment by vladan-me** [comment]
Hi guys, since there were no words from authors for a long time, I've decided to start a [fork](https://github.com/vladan-me/vafpress-framework/tree/develop) of this framework. 
@Zackio I'd be happy to merge your fixes from your fork, assuming it's tested enough. :)

**Comment by wpsoul** [comment]
I'm glad to see that someone continue to develop framework, because, looks like author stop to do this. I will follow new fork.

**Comment by ghost** [comment]
Good to see this is being forked. Why not make a new name for it, and modify the heck out of it to make it much more user friendly, make the UI take advantage of full width of the screen as well while you are at it.

**Comment by Zackio** [comment]
Good to see interest in keeping this project going, I've been using my fix for sometime with no further problems. Would be good to share any other additions to the framework. I have made fields for taxonomies and preview button added as custom fields.

**Comment by wpsoul** [comment]
Fields for taxonomy? Can you contact with vladan-me and add your additions to his fork?

**Comment by Zackio** [comment]
works like this:

```
    array(
        'type' => 'taxonomy',
        'heading' => 'Categories',
        'taxonomy' => 'category',
        'param_name' => 'terms',
        ),
```

Displays checkboxes for each category so a user can select custom post types by category

Currently I have then outside of the framework and included, I'll set them up inside the framework so I can share them.

**Comment by vladan-me** [comment]
@Zackio Once you are ready, please make an issue for taxonomy fields, explain how it works and what is done in code. Also, all other improvements that you have and are still not around, please add so I can take a look on it and add it. Thanks!

**Comment by Zackio** [comment]
Sure, I'll do that soon as I can.

**Comment by Zackio** [comment]
I've uploaded my extra fields:

https://github.com/Zackio/Vafpress-Extra-Fields

It's working and well tested but I need to check my instructions are all correct
